analogin.py

```diff
@@ -215,4 +236,59 @@ def data_out(self) -> str:
             self.data_string = TCP.format_data('AI/dimensions', shape_str) + \
                 TCP.format_data('AI/data', data_bytes)
                 
-            return self.data_string
\ No newline at end of file
+            return self.data_string
+            
+            
+    def start(self):
+        """
+        Start the task
+        """
+
+        if not (self.stop_connections or self.reset_connection) and self.enable:
+            try:
+                self.task.start()
+                
+            except DaqError as e:
+                # end the task nicely
+                self.stop()
+                self.close()
+                msg = '\n AnalogInput failed to start task'
+                raise HardwareError(self, task, message=msg)
+
+            except DaqWarning as e:
+                self.logger.warning(str(e.message))
+            
+
+    def stop(self):
+        """
+        Stop the task
+        """
+        
+        if self.task != None:
+            try:
+                self.task.stop()
+                
+            except DaqError as e:
+                self.close()
+                msg = '\n AnalogInput failed to stop current task'
+                raise raise HardwareError(self, task, message=msg)
+
+            except DaqWarning as e:
+                self.logger.warning(str(e.message))
```

Comment:
seems like this code block is repeated a few times in this file (except for the message). Could you create a function that runs the function you want and takes care of this error handling?

Reply:
yes, probably. i do like the transparency of this though, so I think it's okay for now at least.